frame/democracy/src/lib.rs

```diff
@@ -1414,12 +1552,13 @@ impl<T: Config> Pallet<T> {
 		if let Some((proposal, threshold)) = <NextExternal<T>>::take() {
```

Comment:
Consider early return instead of if..else.

Reply:
I see you point, but I would rather do not touch the audited code if not necessary.
This PR I think still can pass as 'nice to audit' instead 'must have'